utils/azureCredentialsManager.ts

```diff
@@ -0,0 +1,138 @@
+import { SubscriptionClient, ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { AzureAccount } from '../typings/azure-account.api';
+import { ServiceClientCredentials } from 'ms-rest';
+import { AsyncPool } from '../utils/asyncpool';
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import * as ContainerModels from '../node_modules/azure-arm-containerregistry/lib/models';
+import { ResourceGroup, ResourceGroupListResult } from "azure-arm-resource/lib/resource/models";
+import { MAX_CONCURRENT_SUBSCRIPTON_REQUESTS } from './constants';
+
+/* Singleton for facilitating communication with Azure account services by providing extended shared
+  functionality and extension wide access to azureAccount. Tool for internal use.
+  Authors: Esteban Rey L, Jackson Stokes
+*/
+
+export class AzureCredentialsManager {
+
+    //SETUP
+    private static _instance: AzureCredentialsManager = new AzureCredentialsManager();
+    private azureAccount: AzureAccount;
+
+    constructor() {
+        if (AzureCredentialsManager._instance) {
+            throw new Error("Error: Instantiation failed: Use SingletonClass.getInstance() instead of new.");
+        }
+        AzureCredentialsManager._instance = this;
+    }
+
+    public static getInstance(): AzureCredentialsManager {
+        return AzureCredentialsManager._instance;
+    }
+
+    //This function has to be called explicitly before using the singleton.
+    public setAccount(azureAccount) {
+        this.azureAccount = azureAccount;
+    }
+
+    //GETTERS
+    public getAccount() {
+        if (this.azureAccount) return this.azureAccount;
+        throw ('Azure account is not present, you may have forgotten to call setAccount');
+    }
+
+    public getFilteredSubscriptionList(): SubscriptionModels.Subscription[] {
+        return this.getAccount().filters.map<SubscriptionModels.Subscription>(filter => {
+            return {
+                id: filter.subscription.id,
+                session: filter.session,
+                subscriptionId: filter.subscription.subscriptionId,
+                tenantId: filter.session.tenantId,
+                displayName: filter.subscription.displayName,
+                state: filter.subscription.state,
+                subscriptionPolicies: filter.subscription.subscriptionPolicies,
+                authorizationSource: filter.subscription.authorizationSource
+            };
+        });
+    }
+
+    public getContainerRegistryManagementClient(subscription: SubscriptionModels.Subscription): ContainerRegistryManagementClient {
```

Comment:
Looks like this method will be used internally and can be made `private`?

Reply:
That method is meant to be for external use, maybe providing a different name/ providing additional filtering and labeling it getSubscriptions where this particular filtering is default would be better?